### PR TITLE
generate: drop an all-ones attention mask from model_kwargs

### PR DESCRIPTION
**Pull request by SunMarc**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48803) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48803)

Same as https://github.com/huggingface/transformers/pull/47951 but i try to remove the `attention_mask` from `model_kwargs`.

### CODE REVIEW

File: src/transformers/generation/utils.py [hunk 1]
```diff
@@ -2766,6 +2767,30 @@ def generate(
...
+        generation_config._inputs_hold_full_sequence = (
+            length_mask is not None and input_ids.shape[1] == length_mask.shape[1]
+        )
+        decoding_name = GENERATION_MODES_MAPPING[generation_mode]
+        uses_default_decoding_loop = (
+            "/" not in decoding_name
+            and decoding_method is getattr(GenerationMixin, decoding_name)
+            # Candidate generators slice `model_kwargs["attention_mask"]` themselves
+            and generation_mode != GenerationMode.ASSISTED_GENERATION
+        )
```
Comment (SunMarc) [inline src/transformers/generation/utils.py]: i had the restrict a bit the fix since most methods still uses the attn masks but those can maybe be fixed

File: src/transformers/generation/utils.py [hunk 2]
```diff
@@ -2700,6 +2711,27 @@ def generate(
         if not kwargs_has_position_ids and accepts_position_ids and not self.config.is_encoder_decoder:
             model_kwargs["position_ids"] = self._prepare_position_ids_for_generation(inputs_tensor, model_kwargs)
```
Comment (SunMarc) [inline src/transformers/generation/utils.py:2713]: we could delete the attention_mask even earlier but we will have to pass a new arg to `_prepare_position_ids_for_generation` : `generation_config._mask_length` so that we can create correctly the `positions_ids`. Note that we would have to update the models where `_prepare_position_ids_for_generation` is overwritten . Also some models do `self.model.get_rope_index(inputs_tensor, **model_kwargs)` where they use the attn mask

**Comment by SunMarc** [comment]
closed in favor of https://github.com/huggingface/transformers/pull/48814 !
